### DISCUSSION TITLE
How to get confidence scores when using extended_summary with MeanAveragePrecision?

### DISCUSSION THREAD

**Discussion by bradsteiner-sd** [Lightning-AI/torchmetrics#2201] [CompVision] [answered]
Hi, I am using the `extended_summary` flag with `MeanAveragePrecision()` and getting the precision values for each of the recall thresholds. However, I would like to know what confidence threshold is used at each of those recall points. In other words, how do I get the operating points for each location along the precision-recall curve? Is this possible?

**Comment by bradsteiner-sd** [comment]
I can submit a PR that makes this change, I have updated my local code with this improvement

**Comment by ocharles** [comment]
I believe this got fixed in https://github.com/Lightning-AI/torchmetrics/issues/2202 - right?

**Comment by Borda** [comment, marked as answer]
@ocharles thanks for flagging PR #2202. To expand on this for anyone landing here:

As of TorchMetrics v1.9.0, `MeanAveragePrecision` supports `extended_summary=True` which returns additional tensors including IoU values, precision, recall, and scores at each evaluation point:
```python
from torchmetrics.detection import MeanAveragePrecision

metric = MeanAveragePrecision(
    iou_type="bbox",
    extended_summary=True,
)
metric.update(preds, targets)
result = metric.compute()

# Standard keys:
# result["map"], result["map_50"], result["map_75"], ...

# Extended summary keys:
# result["ious"]       — dict of IoU matrices per image pair
# result["precision"]  — precision values at each (iou_threshold, class, area, max_det) combination
# result["recall"]     — recall values at each combination
# result["scores"]     — confidence score thresholds corresponding to precision/recall
```

The `scores` tensor follows the COCO evaluation structure: shape `(T, K, A, M)` where T = IoU thresholds, K = classes, A = area ranges (all, small, medium, large), M = max detection thresholds. You can slice it to get the confidence score at a specific operating point.

**Example — getting confidence threshold at 0.5 IoU, for class 0, all areas, max 100 dets:**
```python
score_threshold = result["scores"][0, 0, 0, 2]  # T=0, K=0, A=0(all), M=2(100 dets)
```

If `scores` is not present in your output, make sure you're on v1.8+ and have `extended_summary=True`.

Docs: [MeanAveragePrecision](https://lightning.ai/docs/torchmetrics/stable/detection/mean_average_precision.html)
